backend/server/routers/courses.py

```diff
@@ -129,23 +134,36 @@ def search(string):
           “COMP1531”: “SoftEng Fundamentals, 
             ……. }
     """
-    # TODO: is regex search really something we want?
-    # malicious regex can cause DOS depending on regex implementation
-    # Would fuzzy search be better?
-    pat = re.compile(r"{}".format(string), re.I)
-    code_query = list(coursesCOL.find({"code": {"$regex": pat}}))
-    title_query = list(coursesCOL.find({"title": {"$regex": pat}}))
-
-    if not code_query and not title_query:
-        for year in sorted(ARCHIVED_YEARS, reverse=True):
-            code_query = list(archivesDB[str(year)].find({"code": {"$regex": pat}}))
-            title_query = list(archivesDB[str(year)].find({"title": {"$regex": pat}}))
-            if code_query or title_query:
-                break
+    # TODO: consider ways to search legacy courses
+    specialisations = list(userData.specialisations.keys())
+    major = None
+    minor = None
 
-    return {
-        course["code"]: course["title"] for course in chain(code_query, title_query)
-    }
+    # TODO: can you have a minor with no major in Circles?
+    if len(specialisations) == 1:
+        major = specialisations[0]
+    elif len(specialisations) == 2:
+        major = specialisations[0]
+        minor = specialisations[1]
+
+    structure = getStructure(userData.program, major, minor)['structure']
```

Comment:
this is an annoying comment from me - but im curious if you can also do `getStructure(userData.progam, *specialisations)` I think that works and gets rid of the messiness from earlier.